### PR TITLE
feat(github-actions): add support for version tag prefixes

### PR DESCRIPTION
**Pull request by art-shutter**
## Changes

This adds support of tags with prefixes (like `whatever-prefix-v1.2.0`) to github-actions manager.

## Context

Support of version tags with prefixes is useful for reusable actions contained in monorepos and versioned separately. Example: a monorepo containing three actions: "action-one", "action-two", "action-three" all being released with different version tags, "action-one-v1.3.0", "action-two-v2.1.0", "action-three-v1.3.1". Renovatebot should be able to parse such tags and follow them when they are pinned.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

When reviewing please have a look into how the new accepted format of version tags gets interpreted in terms of semver etc.

### CODE REVIEW

File: lib/modules/manager/github-actions/__fixtures__/workflow_4.yml [hunk 1]
```diff
@@ -19,3 +19,4 @@ jobs:
       - uses: actions/checkout@689fcce700ae7ffc576f2b029b51b2ffb66d3abd # v2.1.0 additional comment
       - uses: actions/checkout@689fcce700ae7ffc576f2b029b51b2ffb66d3abd # ratchet:actions/checkout@v2.1.0
       - uses: actions/checkout@689fcce700ae7ffc576f2b029b51b2ffb66d3abd # ratchet:exclude
+      - uses: actions-runner-controller/execute-assert-arc-e2e@f1d7c52253b89f0beae60141f8465d9495cdc2cf # actions-runner-controller-0.23.5
```
Comment (Zebradil) [inline lib/modules/manager/github-actions/__fixtures__/workflow_4.yml:22]: Does it make sense to also add a test case with digits in the textual part of the tag? For example: `base64-encoder-v1.11.3`.

File: lib/modules/manager/github-actions/readme.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
 The `github-actions` manager extracts dependencies from GitHub Actions workflow and workflow template files.
 It can also be used for Gitea and Forgejo Actions workflows as such are compatible with GitHub Actions workflows.
 
-If you like to use digest pinning but want to follow the action version tag, you can use the following sample:
+If you like to use digest pinning but want to follow the action version tag, you can use the sample below.
+The version tag can be in the format `(prefix-)(v)1.0.0`, where `prefix` and `v` are optional and `1.0.0` is the version number.
```
Comment (rarkins) [inline lib/modules/manager/github-actions/readme.md:5]: This sentence is pretty hard to understand, so best to add to the examples below to clarify.

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/github-actions/extract.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ import type { Workflow } from './types';
 
 const dockerActionRe = regEx(/^\s+uses\s*: ['"]?docker:\/\/([^'"]+)\s*$/);
 const actionRe = regEx(
-  /^\s+-?\s+?uses\s*: (?<replaceString>['"]?(?<registryUrl>https:\/\/[.\w-]+\/)?(?<depName>[\w-]+\/[.\w-]+)(?<path>\/.*)?@(?<currentValue>[^\s'"]+)['"]?(?:\s+#\s*(((?:renovate\s*:\s*)?(?:pin\s+|tag\s*=\s*)?|(?:ratchet:[\w-]+\/[.\w-]+)?)@?(?<tag>v?\d+(?:\.\d+(?:\.\d+)?)?)|(?:ratchet:exclude)))?)/,
+  /^\s+-?\s+?uses\s*: (?<replaceString>['"]?(?<registryUrl>https:\/\/[.\w-]+\/)?(?<depName>[\w-]+\/[.\w-]+)(?<path>\/.*)?@(?<currentValue>[^\s'"]+)['"]?(?:\s+#\s*(((?:renovate\s*:\s*)?(?:pin\s+|tag\s*=\s*)?|(?:ratchet:[\w-]+\/[.\w-]+)?)@?(?<tag>([\w-]*-)?v?\d+(?:\.\d+(?:\.\d+)?)?)|(?:ratchet:exclude)))?)/,
```
Comment (viceice) [inline lib/modules/manager/github-actions/extract.ts:18]: this regex should be split into multiple parts , it becomes unmaintainable. can be done in separate PR.

File: lib/modules/manager/github-actions/readme.md [hunk 4]
```diff
@@ -15,8 +15,10 @@ jobs:
       - uses: actions/checkout@3df4ab11eba7bda6032a0b82a6bb43b11571feac # v4.0.0
 ```
 
-Renovate will update the commit SHA but follow the GitHub tag you specified.
+Renovate will update the commit SHA according to the GitHub tag you specified.
 Renovate can update digests that use SHA1 and SHA256 algorithms.
+The GitHub tag is in the format of `(prefix-)(v)1.0.0`, where `prefix` and `v` are optional and `1.0.0` is the version number.
```
Comment (viceice) [inline lib/modules/manager/github-actions/readme.md:20]: `v1` and `1.1` are also valid tags and are already captured

Comment (art-shutter) [reply lib/modules/manager/github-actions/readme.md:20]: fixed

**Review by viceice — APPROVED** [review]

**Comment by milpog** [comment]
Is there similar functionality available for git-tags and/or gtihub-tags?

**Comment by viceice** [comment]
please open a new discussion

**Comment by milpog** [comment]
> please open a new discussion

Sure. I think [I found already opened discussion for that](https://github.com/renovatebot/renovate/discussions/24876).

**Pull request merged**